### PR TITLE
fix(CX-1656): Prevent facebook auth for users without email

### PR DESCRIPTION
**Pull request by edez1t**
The type of this PR is: **Bugfix**

This PR resolves [CX-1656]

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

[CX-1574]: https://artsyproduct.atlassian.net/browse/CX-1574

[CX-1656]: https://artsyproduct.atlassian.net/browse/CX-1656

### CODE REVIEW

File: src/lib/store/AuthModel.ts [hunk 1]
```diff
@@ -268,11 +268,16 @@ export const getAuthModel = (): AuthModel => ({
 
       const responseFacebookInfoCallback = async (
         error: { message: string },
-        facebookInfo: { email: string; name: string }
+        facebookInfo: { email?: string; name: string }
       ) => {
         if (error) {
           reject(`Error fetching facebook data: ${error.message}`)
         } else {
+          if (!facebookInfo.email) {
```
Comment (olerichter00) [inline src/lib/store/AuthModel.ts:276]: I think we can use `return reject(...)` in both cases and get rid of the `else`.

File: src/lib/store/AuthModel.ts [hunk 2]
```diff
@@ -268,11 +268,16 @@ export const getAuthModel = (): AuthModel => ({
...
+            reject("You are not using email on your Facebook account. Try sign up using credentials.")
```
Comment (olerichter00) [inline src/lib/store/AuthModel.ts:277]: Let's we ask Jonathan for the wording of the message.

**Review by olerichter00 — APPROVED** [review]
LGTM! 
Just two small comments to be addressed.

File: src/lib/store/AuthModel.ts (hunk 2 above)
Comment (edez1t) [reply src/lib/store/AuthModel.ts]: done

File: src/lib/store/AuthModel.ts (hunk 1 above)
Comment (edez1t) [reply src/lib/store/AuthModel.ts]: done

**Comment by edez1t** [comment]
#squashongreen

**Pull request merged**